common/dns.go

```diff
@@ -0,0 +1,89 @@
+package common
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/giantswarm/clustertest/pkg/application"
+	"github.com/giantswarm/clustertest/pkg/wait"
+	"github.com/miekg/dns"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+func runDNS() {
+	Context("dns", func() {
+		var (
+			resolver *net.Resolver
+			values   *application.DefaultAppsValues
+		)
+
+		BeforeEach(func() {
+			values = &application.DefaultAppsValues{}
+			defaultAppsName := fmt.Sprintf("%s-default-apps", Cluster.Name)
+			err := Framework.MC().GetHelmValues(defaultAppsName, Cluster.Namespace, values)
+			Expect(err).NotTo(HaveOccurred())
+
+			resolver = &net.Resolver{
+				PreferGo:     true,
+				StrictErrors: true,
+				Dial: func(ctx context.Context, network, address string) (net.Conn, error) {
+					d := net.Dialer{
+						Timeout: time.Millisecond * time.Duration(10000),
+					}
+					return d.DialContext(ctx, "udp", "8.8.4.4:53")
+				},
+			}
+		})
+
+		It("sets up the api DNS records", func() {
```

Comment:
just FYI: as long as no "private" Clusters got created in `CAPZ`, everything is fine.
But if the WC is a private one, depending where these tests are run, if it's in the MC itself, the Record might be a different one - https://github.com/giantswarm/dns-operator-azure#architecture